### ISSUE TITLE
[harfbuzz] build failure

### ISSUE THREAD

**Issue by Kicer86** [microsoft/vcpkg#18907]
**Host Environment**
- OS: Windows
- Compiler: gcc.exe (Rev5, Built by MSYS2 project) 10.3.0

**To Reproduce**
vcpkg.exe install harfbuzz:x64-mingw-dynamic

**Failure logs** 
Output:

```
Computing installation plan...
The following packages will be built and installed:
    harfbuzz[core]:x64-mingw-dynamic -> 2.8.1
Detecting compiler hash for triplet x64-mingw-dynamic...
Could not locate cached archive: C:\Users\michal\AppData\Local\vcpkg\archives\6c\6c257d0c023ea60dcd88b5135be81e09f05ed9ac.zip
Starting package 1/1: harfbuzz:x64-mingw-dynamic
Building package harfbuzz[core]:x64-mingw-dynamic...
-- Using community triplet x64-mingw-dynamic. This triplet configuration is not guaranteed to succeed.
-- [COMMUNITY] Loading triplet configuration from: C:\vcpkg\triplets\community\x64-mingw-dynamic.cmake
-- Using C:/vcpkg/downloads/harfbuzz-harfbuzz-b37f03f16b39d397a626f097858e9ae550234ca0.tar.gz
-- Cleaning sources at C:/vcpkg/buildtrees/harfbuzz/src/e550234ca0-59b6eda119.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source C:/vcpkg/downloads/harfbuzz-harfbuzz-b37f03f16b39d397a626f097858e9ae550234ca0.tar.gz
-- Applying patch 0001-circumvent-samefile-error.patch
-- Applying patch 0002-fix-uwp-build.patch
-- Using source at C:/vcpkg/buildtrees/harfbuzz/src/e550234ca0-59b6eda119.clean
-- Configuring x64-mingw-dynamic
-- Using msys root at C:/vcpkg/downloads/tools/msys2/aa5af7b2aa7e90e8
-- Configuring x64-mingw-dynamic-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:105 (message):
    Command failed: C:/vcpkg/downloads/tools/python/python-3.9.5-x64/python.exe C:/vcpkg/downloads/tools/meson/meson-753954be868ed78b3e339e8811fd1d29eb2af237/meson.py -Dicu=disabled -Dgraphite=disabled -Dcoretext=disabled -Dglib=disabled -Dgobject=disabled -Dfreetype=enabled -Dcairo=disabled -Dfontconfig=disabled -Dintrospection=disabled -Ddocs=disabled -Dtests=disabled -Dbenchmark=disabled --buildtype plain --backend ninja --wrap-mode nodownload --native C:/vcpkg/buildtrees/harfbuzz/meson-nativ-x64-mingw-dynamic.log --default-library shared --libdir lib --native C:/vcpkg/buildtrees/harfbuzz/meson-nativ-x64-mingw-dynamic-debug.log -Ddebug=true --prefix C:/vcpkg/packages/harfbuzz_x64-mingw-dynamic/debug --includedir ../include -Dcmake_prefix_path=['C:/vcpkg/installed/x64-mingw-dynamic/debug','C:/vcpkg/installed/x64-mingw-dynamic','C:/vcpkg/installed/x64-mingw-dynamic/share'] C:/vcpkg/buildtrees/harfbuzz/src/e550234ca0-59b6eda119.clean
    Working Directory: C:/vcpkg/buildtrees/harfbuzz/x64-mingw-dynamic-dbg
    Error code: 1
    See logs for more information:
      C:\vcpkg\buildtrees\harfbuzz\config-x64-mingw-dynamic-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_configure_meson.cmake:485 (vcpkg_execute_required_process)
  ports/harfbuzz/portfile.cmake:48 (vcpkg_configure_meson)
  scripts/ports.cmake:139 (include)


Error: Building package harfbuzz:x64-mingw-dynamic failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: harfbuzz:x64-mingw-dynamic
  Vcpkg version: 2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9

Additionally, attach any relevant sections from the log files above.
```

C:\vcpkg\buildtrees\harfbuzz\config-x64-mingw-dynamic-dbg-out.log:
```
DEPRECATION: c_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: cpp_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: c_link_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
DEPRECATION: cpp_link_args in the [properties] section of the machine file is deprecated, use the [built-in options] section.
WARNING: Recommend using either -Dbuildtype or -Doptimization + -Ddebug. Using both is redundant since they override each other. See: https://mesonbuild.com/Builtin-options.html#build-type-options
The Meson build system
Version: 0.58.0
Source dir: C:\vcpkg\buildtrees\harfbuzz\src\e550234ca0-59b6eda119.clean
Build dir: C:\vcpkg\buildtrees\harfbuzz\x64-mingw-dynamic-dbg
Build type: native build
WARNING: Unknown options: "fontconfig"
The value of new options can be set with:
meson setup <builddir> --reconfigure -Dnew_option=new_value ...
Project name: harfbuzz
Project version: 2.8.1

..\src\e550234ca0-59b6eda119.clean\meson.build:1:0: ERROR: Unsupported linker, only bfd, gold, and lld are supported, not C:/Users/michal/tools/msys64/mingw64/bin/ld.exe.

A full log can be found at C:\vcpkg\buildtrees\harfbuzz\x64-mingw-dynamic-dbg\meson-logs\meson-log.txt
```

**Comment by Kicer86** [comment]
I am aware that `x64-mingw-dynamic` triplet comes from community, yet I hope this issue could be easily fixed.

**Comment by autoantwort** [comment]
Would be fixed by #18927 :)

**Comment by FrankXie05** [comment]
@Kicer86 The fixed PR has been merged, Please use the latest version of vcpkg and reopen it if the problem still exists
